server/py/services/api/crud/model_monitoring/deployment.py

```diff
@@ -1347,60 +1360,43 @@ async def create_model_endpoints(self, function: dict, function_name: str):
                 mm_constants.EventFieldType.SAMPLING_PERCENTAGE, 100
             ),
         )  # model endpoint, creation strategy, model path
-        router_model_endpoints_instructions: list[
+        semaphore = Semaphore(50)  # Limit concurrent tasks
```

Comment:
Would be nice to add a TODO about replacing this parallelism mechanism with batching, by refactoring `_create_model_endpoint_limited`.

I'm also a bit concerned that while this reduces the runtime, it does so at the expense of other mlrun-api work (is it still responsive while this is going on?).

Reply:
We do run batch of 500..

Reply:
Right, wrong place for that comment. The issue is that `create_model_endpoints` then takes that batch and for loops over it, doing IO on each row separately.